geps/gep-1619.md

```diff
@@ -62,14 +69,131 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
+session.
+
+Here an example implementation of a gateway initiating a session through cookies:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B-->>-G: Response
+    G->>G: Add set-cookie header
+    G-->>-C: Response<br>[set-cookie]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header.<br>May include other set-cookie<br>headers from backend.
+    C->>C: Create Cookie(s)<br>from set-cookie header(s)
+    Note right of C: [cookie] indicates a request<br> with one or more cookies
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup of<br>server using cookie value
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+#### Backend Initiated Session Example
+
+Continuing with the cookie example, when dealing with backend-initiated sessions, the process becomes somewhat more
+complex. For cookie-based session persistence, the gateway needs store a value within a cookie containing a server
+identifier. This identifier can be then used as a reference to maintain a persistent session to a specific backend.
+There are several approaches a gateway could use in this situation to achieve session persistence:
+
+1. Modify the existing cookie's value
+2. Insert an additional cookie
+3. Prefix the existing cookie
+
+Alternatively, if the backend is not configured for session persistence, the gateway should refrain from modifying or
+inserting a cookie. In this situation, the gateway should remain passive and simply forward the `set-cookie` header as
+it is.
+
+Refer to the [Cookie Rewriting](#cookie-rewriting) section of the API for implementation guidance.
+
+Here's an example implementation of a backend initiating a session and the gateway modifies the cookie:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B->>B: Add set-cookie<br>header
+    B-->>-G: Response<br>[set-cookie]
+    G->>G: Modify set-cookie<br>header per configuration
+    G-->>-C: Response<br>[set-cookie*]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header<br>[set-cookie*] indicates a response<br>with a MODIFIED set-cookie header
+    C->>C: Create Cookie<br>from set-cookie header
+    Note right of C: [cookie] indicates a request<br>or response with a cookie
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup<br>of server using cookie value
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+#### Global Load Balancer Initiated Session Example
+
+In a more complex architecture, a global load balancer may initiate a session in order to maintain persistent
```

Comment:
I think the confusion is around "GLB maintain a session with a regional LB" - it doesn't have to have a session, just need to send traffic to the same region, by setting a cookie for example. Depending on implementation it may send traffic to a close region if the LB in the target region is overloaded/down, but probably less common.
But the GLB doesn't initiate 'sessions' in the same sense as the regional LB - can still load balance base across all instances of the regional LB. 

When I read the text I was confused by the 'initiate a session', not sure what would be a better word for GLB sending to a region.